src/Form/RenderFileTab.tsx

```diff
@@ -30,7 +29,8 @@ export default function RenderFileTab(step: Step, steps: Array<Step>, setSteps:
   }
 
   const displayFilename = (filename: string) => {
-    return filename.length > 12 ? filename.split('.')[0].substring(0, 12) + '...' + filename.split('.')[1] : filename
+    let filenameSplit = filename.split('.')
+    return filenameSplit[0].length > 12 ? filenameSplit[0].substring(0, 12) + '...' + filenameSplit[1] : filename
```

Comment:
This breaks if given a filename like `model.tar.gz` (or any string with multiple dots).  Maybe rewrite as:

```
const displayFilename = (filename: string) => {
  const parts = filename.split('.')
  const ext = parts.pop()
  const base = parts.join('.')

  return base.length > 12 ? `${base}...${ext}` : filename
}

Reply:
Much better, changed!